src/run/run.c

```diff
@@ -1020,31 +1024,27 @@ static int start_transient_service(
                         pty_forward_set_handler(c.forward, pty_forward_handler, &c);
                 }
 
-                if (arg_wait) {
-                        _cleanup_free_ char *path = NULL;
-                        const char *mt;
 
-                        path = unit_dbus_path_from_name(service);
-                        if (!path)
-                                return log_oom();
+                path = unit_dbus_path_from_name(service);
+                if (!path)
+                        return log_oom();
 
-                        mt = strjoina("type='signal',"
-                                      "sender='org.freedesktop.systemd1',"
-                                      "path='", path, "',"
-                                      "interface='org.freedesktop.DBus.Properties',"
-                                      "member='PropertiesChanged'");
-                        r = sd_bus_add_match(bus, &c.match, mt, on_properties_changed, &c);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to add properties changed signal.");
+                mt = strjoina("type='signal',"
+                              "sender='org.freedesktop.systemd1',"
+                              "path='", path, "',"
+                              "interface='org.freedesktop.DBus.Properties',"
+                              "member='PropertiesChanged'");
```

Comment:
Note that the whole piece is enclosed in a big `if (arg_wait || master >= 0) {` check already.

`master >= 0` is a different way for checking `arg_pty`, hence this is already enclosed in an `arg_wait || arg_pty` check, and has no effect if neither is specified. The only change really is that if arg_pty alone is specified we'll now also watch for the signal, which we previously only did on arg_wait...

Hope that makes sense?